custom/enikshay/case_utils.py

```diff
@@ -121,3 +130,17 @@ def get_adherence_cases_between_dates(domain, person_case_id, start_date, end_da
     ]
 
     return open_pertinent_adherence_cases
+
+
+def update_case(domain, case_id, updated_properties, external_id=None):
+    kwargs = {
+        'case_id': case_id,
+        'update': updated_properties,
+    }
+    if external_id is not None:
```

Comment:
@proteusvacuum Why do we need this check here? No strong but I think would be good to keep `update_case` generic instead of over writing this specifically for this update.
How i kept this earlier was to mark this as a failure since nikshay_id/external_id should be needed, save the response for debugging and not update the case. Hence, handle this from where it is called.

I am not sure but saving None value is an issue in some way? Just checking for the need to remove `external_id` if its none.